src/test/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentReportItemTest.java

```diff
@@ -0,0 +1,93 @@
+    
+    CreateAccountingDocumentReportItem reportItem;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        reportItem = new CreateAccountingDocumentReportItem(FILE_NAME);
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, StringUtils.EMPTY));
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DI, TWO_SPACES));
+        reportItem.getDocumentsSuccessfullyRouted().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, TWO_SPACES));
+        reportItem.getDocumentsSuccessfullyRouted().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DI, StringUtils.EMPTY));
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        reportItem = null;
+    }
+    
+    private CreateAccountingDocumentReportItemDetail createCreateAccountingDocumentReportItemDetail(String documentType, String warningMessage) {
+        CreateAccountingDocumentReportItemDetail detail = new CreateAccountingDocumentReportItemDetail();
+        detail.setDocumentType(documentType);
+        detail.setWarningMessage(warningMessage);
+        return detail;
+    }
+    
+    private void addWarningDetails() {
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, DEFAULT_WARNING_MESSAGE));
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DI, DEFAULT_WARNING_MESSAGE));
+        reportItem.getDocumentsSuccessfullyRouted().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, DEFAULT_WARNING_MESSAGE));
+    }
+
+    @Test
+    public void testGetAllDocumentDetails() {
+        assertEquals(4, reportItem.getAllDocumentDetails().size());
+    }
+    
+    @Test
+    public void testGetDocumentDetailsWithWarningsNoWarnings() {
+        assertEquals(0, reportItem.getDocumentDetailsWithWarnings().size());
+    }
+    
+    @Test
+    public void testGetDocumentDetailsWithWarningsWithWarnings() {
+        addWarningDetails();
+        assertEquals(3, reportItem.getDocumentDetailsWithWarnings().size());
+    }
+    
+    @Test
+    public void testGetDocumentTypeWarningMessmageCountMapNoWarnings() {
+        Map<String, Integer> docTypeCountMap = reportItem.getDocumentTypeWarningMessmageCountMap();
+        assertEquals(0, docTypeCountMap.size());
+    }
+    
+    @Test
+    public void testGetDocumentTypeWarningMessmageCountMapWithWarnings1() {
+        addWarningDetails();
+        Map<String, Integer> docTypeCountMap = reportItem.getDocumentTypeWarningMessmageCountMap();
+        assertEquals(2, docTypeCountMap.size());
+        assertEquals(1, docTypeCountMap.get(DOCUMENT_TYPE_DI));
+        assertEquals(2, docTypeCountMap.get(DOCUMENT_TYPE_DV));
+    }
+    
+    @Test
+    public void testGetDocumentTypeWarningMessmageCountMapWithWarnings2() {
```

Comment:
Another method with the "Messmage" typo.

Reply:
ok